### ISSUE TITLE
[Feature proposal] Support opening files specifying line

### ISSUE THREAD

**Issue by Dune-jr** [richrd/suplemon#248]
There is this Sublime Text feature that you can open a file like this:
```
subl src/engine/client/input.cpp:53
```
and it will open `input.cpp` directly at line `53`. Very useful when copying `grep -n` results, for example. 

It even supports this format for columns:
```
subl src/engine/client/input.cpp:53:2
```

**Comment by richrd** [comment]
I really like the idea here, and I think that should be supported.
What should happen if there is a file with the name specified e.g. `file:12:34`?

I guess the logic could be the following:
1) first check whether the raw filename exists (if e.g. `file:12:34` does then just load it)
2) if not trim off everything from the end after the first colon
3) if the file path matches then load it and apply the relevant cursor position after said first colon

**Comment by Dune-jr** [comment]
> I really like the idea here, and I think that should be supported.
> What should happen if there is a file with the name specified e.g. `file:12:34`?

The best bet to me would be to replicate ST's behavior.

I cannot test this on Windows unfortunately, because `:` is forbidden here :)

I can tell you however that `subl test:12` (test does not exist): opens a new file named `test`

> I guess the logic could be the following:
> 
> 1. first check whether the raw filename exists (if e.g. `file:12:34` does then just load it)
> 2. if not trim off everything from the end after the first colon
> 3. if the file path matches then load it and apply the relevant cursor position after said first colon

Yeah that sounds good and doesn't limit the range of files suplemon can open.

**Comment by caph1993** [comment]
I checked the behavior of sublime in case you want to replicate it. I created three files: `test`, `test:2` and `test:2:2`.

- Running `subl test:2` opens `test:2`.
- Running `subl test:2:2` opens `test:2:2`.
- Running `subl test:2:1` opens `test` at line 2, char 1. Why not `test:2` at line 1? don't know.
- Running `subl test:2:2:1` opens `test:2` at line 2, char 1. Why not `test:2:2` at line 1? don't know.
- Running `subl test:2:2:2:2` opens `test:2:2` at line 2, char 2.

The logic seems to be:

1. Try matching `filename` (If the whole string matches a file, open that file).
2. Try matching `filename:line_num:char_num`.
3. Try matching `filename:line_number`.

I thought 2 and 3 were checked in the opposite order because it makes more sense to me, but it seems that sublime does it in order 1-2-3.

**Comment by richrd** [comment]
Thanks for the tests, I appreciate it! I've already worked on this and I'll try to make it as close to sublime as possible. This will be coming in the next release.

**Comment by richrd** [comment]
I just released 0.2.0 which has this feature.

**Comment by Dune-jr** [comment]
Thanks :)
